Dockerization + Zeit Deployment

atomdmac opened the pull request:
# Goal of PR
Add support for Drone builds and deployment to Zeit Now on push to `master`.

# TODO
 - [x] Make sure pushes to PRs don't double-trigger build pipelines
 - [x] Add test stage
 - [ ] ~~Determine if Google login breakage is the fault of this PR or not.~~

atomdmac committed:
fix: Downgrade min jest version to 22

Apparently, v23 doesn't work on linux?

atomdmac committed:
chore: Add package-lock.json

atomdmac committed:
feat: Add Dockerfile.build

atomdmac committed:
feat: Add Dockerfile for Zeit Now deployment.

atomdmac committed:
feat: Add .drone.yml w/ build + deploy stages.

ryanSN commented on package.json:
```diff
@@ -50,7 +50,7 @@
     "eslint-plugin-vue": "^4.3.0",
     "file-loader": "^1.1.4",
-    "jest": "^23.0.0-alpha.0",
+    "jest": "^22.0.0-alpha.0",
```
I don't think they have a 22 alpha. https://www.npmjs.com/package/jest?activeTab=versions

ryanSN commented on Dockerfile.build:
```diff
@@ -0,0 +1,17 @@
+RUN mkdir /usr/src/app
+WORKDIR /usr/src/app
+ENV PATH /usr/src/app/node_modules:$PATH
+ADD src /usr/src/app/src
```
All works the way you have it. You could probably do a `ADD . /usr/src/app`
As in your NGINX image you are grabbing only the `dist` folder.  It would make your build image slightly bigger as it would grab the tests etc and just be throwing them on the floor. But could allow grabbing any new stuff we add. 🤷‍♂️

ryanSN commented on Dockerfile.build:
```diff
@@ -0,0 +1,17 @@
+ADD .babelrc /usr/src/app
+ADD package.json /usr/src/app
+ADD webpack.config.js /usr/src/app
+RUN npm install && npm run build
```
`npm install --production` unless the build does it.

ryanSN reviewed (approved):
I like 👍.
💯

atomdmac committed:
fix: Use _real_ jest version /facepalm

atomdmac committed:
feat: Add test stage to pipeline.

atomdmac committed:
chore: Add `test` env var to test script.

atomdmac committed:
chore: Include babel-jest explicitly.

Even though this is supposedly included as a dep of `jest`, Drone
won't build without this being explicitly installed.  I'm not sure
why but this seems to fix it.
